### ISSUE TITLE
Is My Model Useless? Roberta tokenization gives <unk> for words in dict.txt 

### ISSUE THREAD

**Issue by hichiaty** [pytorch/fairseq#2581]
#### What is your question?

I pre-trained the standard roberta by encoding my dataset (as pieces) with sentencepiece first using [this script](https://github.com/pytorch/fairseq/blob/master/examples/translation/prepare-iwslt17-multilingual.sh) as suggested in the multilingual translation example. I then used fairseq-preprocess with this command on the sentencepiece encoded textfile:

```
fairseq-preprocess \
    --only-source \
    --trainpref predat/encoded_all_spm.txt \
    --validpref predat/eval_all_spm.txt \
    --destdir data-bin\
    --workers 8
```
This generated a dict.txt file. I then trained/evaluated the model, and the loss was decreasing as expected.

When I try to load the model for use afterwards, encoding any new sentence gives `<unk>` tokens (with id 3).

I load the model with:
```
from fairseq.models.roberta import RobertaModel

roberta = RobertaModel.from_pretrained(
    'fairseq/checkpoints',
    'checkpoint_best.pt',
    'path/to/data-bin'
    )
roberta.eval()
```

After which I try an example sentence, I use Sentencepiece to pre-process then attempt to encode:

```
import sentencepiece as spm
sp_model = spm.SentencePieceProcessor()
sp_model.load("path/to/spm.model")

sentence = ' '.join(sp_model.EncodeAsPieces('example sentence'))
```
This gives `_ex` as the first token, which is in dict.txt.

When I attempt to use `roberta.encode(sentence)` I get the `<unk>` token id for the `_ex` token and most other tokens even though they are in dict.txt.

Anyone know what's going on?

**Comment by hichiaty** [comment]
Solved. Used CamemBERT with both the sentencepiece model and dict.txt

**Comment by yuanenming** [comment]
Hi @hichiaty !

I  have met the same problem. what do you mean by "Used CamemBERT with both the sentencepiece model and dict.txt".

Does it means I should specify the `sentencepiece` model and the `dict.txt` args? or anything else?
🍉
